src/jit/lsraarmarch.cpp

```diff
@@ -792,7 +792,20 @@ void Lowering::TreeNodeInfoInitBlockStore(GenTreeBlk* blkNode)
             // a temporary register to perform the sequence of loads and stores.
             blkNode->gtLsraInfo.internalIntCount = 1;
 
+            if (size >= 2 * REGSIZE_BYTES)
+            {
+                // We will use ldp/stp to reduce code size and improve performance
+                // so we need to reserve an extra internal register
+                blkNode->gtLsraInfo.internalIntCount++;
+            }
+
+            // We can't use the special Write Barrier registers, so exclude them from the mask
+            regMaskTP internalIntCandidates = RBM_ALLINT & ~(RBM_WRITE_BARRIER_DST_BYREF | RBM_WRITE_BARRIER_SRC_BYREF);
+            blkNode->gtLsraInfo.setInternalCandidates(l, internalIntCandidates);
+
+            // If we have a source address we want it in RBM_WRITE_BARRIER_DST_BYREF.
```

Comment:
I think that this comment should read:
` // If we have a dest address we want it in RBM_WRITE_BARRIER_DST_BYREF.`